### ISSUE TITLE
DynamicBindingListBase and async rules

### ISSUE THREAD

**Issue by ajj7060** [MarimerLLC/cslaforum#454]
I think we're hitting an issue with DynamicBindingListBase.  The root objects contained within the list have some async rules, but it looks like the Save method is being called while the object is still busy.

Any suggestions on how to properly handle this?

**Comment by ajj7060** [comment]
@rockfordlhotka @jonnybee Any thoughts on this?

**Comment by jonnybee** [comment]
My best suggestion is to use sync rules. When the user selects another item in the list this will automatically trigger save and is would make a "sluggish" response if the UI must wait for async rules to complete before calling Save. 

Yes, that may provide a sluggish UI when the now sync rules run - but from a functional view this is more "correct" behaviour.

**Comment by rockfordlhotka** [comment]
Thank you @jonnybee, I agree.

@ajj7060 what do you think should be the experience for the user?

**Comment by ajj7060** [comment]
The rule is async b/c it does a remote data portal call, and I believe has to be async to work in Xamarin.  The UI will be sluggish anyway due to a sync Save call blocking the UI thread (as happens today in our WinForms app).  

In our grids we cause the grid to commit changes not on row lost focus but cell lost focus.  To date we've only had this behavior in our WinForms app, but we need the same behavior in our Xamarin app as well.  So probably while the object is busy we'd disable the grid until its not.

Ignoring the UI/usability issues however, it would seem to be a mistake to call Save/SaveAsync on any Csla object while its busy, wouldn't it?

**Comment by rockfordlhotka** [comment]
Yes, but it isn't clear what the alternative would be. Remember that DBLB is designed to interact with the UI _only via established data binding interfaces_, and those interfaces pre-date async _anything_. So there's no established interaction pattern for anything but sync behaviors.

**Comment by rockfordlhotka** [comment]
What grid control are you using in Xamarin?

**Comment by ajj7060** [comment]
We haven't started the Xamarin interface for this particular BO yet, that will be coming in the next phase after we finish the functionality in the WinForms app.  I think we're looking at the grid DevExpress offers for Xamarin.  We found this issue while using the BO in WinForms.

I would think the correct behavior would be to wait until the BO is no longer busy before trying to save.  This has the same issue of making the UI "sluggish," but I think that keeping the BO in a consistent state is important.  

In looking at this a bit more though, it seems like BusinessBase will already throw if Save is called and the object is busy, so I'm not sure why we aren't seeing that exception.  Maybe the DX grid is swallowing it.

Let me look into this a bit more, maybe there's something else going on.
